modules/createHashHistory.js

```diff
@@ -83,7 +87,7 @@ const createHashHistory = (options = {}) => {
   }
 
   const createHref = (path) =>
-    '#' + history.createHref(path)
+    '#' + transformPath(history.createHref(path), true)
```

Comment:
Sounds good. Sorry for sending you on a wild goose chase, but your suggestion makes sense.

I'll have time to to a more detailed review later tonight. Thanks, @tvervest!